pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -283,3 +283,66 @@ func validateSidecarOverrides(ctx context.Context, ts *v1beta1.TaskSpec, trs *v1
 	}
 	return err
 }
+
+// validateResults checks the emitted results type and object properties against the ones defined in spec.
```

Comment:
From the discussion, did you mean you need to bypass the type validations here before applying the results? Can you add a TODO if so?

Reply:
I see. I think the one validation you need to bypass in the validateTaskRunResults() in reconciler taskrun.go. You can close this thread